src/algebra/ordered_pi.lean

```diff
@@ -19,6 +19,34 @@ variables (x y : Π i, f i) (i : I)
 
 namespace pi
 
+/-- The product of a family of ordered commutative monoids is an ordered commutative monoid. -/
+@[to_additive "The product of a family of ordered additive commutative monoids is
+  an ordered additive commutative monoid."]
+instance (ι : Type*) (Z : ι → Type*) [∀ i, ordered_comm_monoid (Z i)] :
+  ordered_comm_monoid (Π i, Z i) :=
+{ mul_le_mul_left := λ f g w h i, mul_le_mul_left' (w i) _,
+  ..(infer_instance : partial_order (Π i, Z i)),
+  ..(infer_instance : comm_monoid (Π i, Z i)) }
+
+/-- The product of a family of canonically ordered monoids is a canonically ordered monoid. -/
+@[to_additive "The product of a family of canonically ordered additive monoids is
+  a canonically ordered additive monoid."]
+instance (ι : Type*) (Z : ι → Type*) [∀ i, canonically_ordered_monoid (Z i)] :
+  canonically_ordered_monoid (Π i, Z i) :=
+{ bot := λ i, 1,
+  bot_le := λ f i, by simp,
```

Comment:
Isn't there an `order_bot` (`pi.order_bot`?) that you can use to populate these fields? If there isn't, there should be.

Reply:
Thanks, I'm using that now.